src/ast/passes/semantic_analyser.cpp

```diff
@@ -2080,6 +2080,13 @@ void SemanticAnalyser::visit(For &f)
   }
   Map &map = static_cast<Map &>(*f.expr);
 
+  if (map.type.IsAvgTy() || map.type.IsHistTy() || map.type.IsLhistTy() ||
+      map.type.IsStatsTy()) {
```

Comment:
Maybe create a helper method (e.g. `IsMapIterableTy`) to simplify management of the (un)supported types?

Reply:
Good idea. Will update.